File: hello-dart/lib/server/audio_server_application.dart
// Main entry point for Dart audio stream server.
// Initializes server components and starts the WebSocket server.
// Matches Java AudioServerApplication functionality.

import 'memory/stream_manager.dart';
import 'memory/memory_pool_manager.dart';
import 'network/audio_websocket_server.dart';
import '../src/logger.dart';

/// Audio server application entry point.
class AudioServerApplication {
  /// Run the audio server application.
  static Future<void> run(List<String> arguments) async {
    // Parse command-line arguments
    int port = 8080;
    String path = '/audio';

    for (int i = 0; i < arguments.length; i++) {
      if (arguments[i] == '--port' && i + 1 < arguments.length) {
        port = int.tryParse(arguments[i + 1]) ?? 8080;
        i++;
      } else if (arguments[i] == '--path' && i + 1 < arguments.length) {
        path = arguments[i + 1];
        i++;
      }
    }

    Logger.info('Starting Audio Server on port $port with path $path');

    // Get singleton instances
    var streamManager = StreamManager.getInstance('cache');
    var memoryPool = MemoryPoolManager.getInstance();

    // Create and start WebSocket server
    var server = AudioWebSocketServer(
      port: port,
      path: path,
      streamManager: streamManager,
      memoryPool: memoryPool,
    );

    // Handle graceful shutdown
    Logger.info('Starting server... Press Ctrl+C to stop.');

    // Start server
    await server.start();
  }
}

void main(List<String> arguments) async {
  await AudioServerApplication.run(arguments);
}

File: hello-kotlin/src/main/kotlin/server/AudioServerApplication.kt
// Main entry point for Kotlin audio stream server.

package server

import server.memory.StreamManager
import server.memory.MemoryPoolManager
import server.network.AudioWebSocketServer

/**
 * Main entry point for audio server application
 */
object AudioServerApplication {
    @JvmStatic
    fun main(args: Array<String>) {
        // Parse command-line arguments
        var port = 8080
        var path = "/audio"
        
        var i = 0
        while (i < args.size) {
            when (args[i]) {
                "--port" -> {
                    if (i + 1 < args.size) {
                        port = args[i + 1].toIntOrNull() ?: 8080
                        i++
                    }
                }
                "--path" -> {
                    if (i + 1 < args.size) {
                        path = args[i + 1]
                        i++
                    }
                }
            }
            i++
        }
        
        Logger.info("Starting Audio Server on port $port with path $path")

        // Get singleton instances
        val streamManager = StreamManager.getInstance("cache")
        val memoryPool = MemoryPoolManager.getInstance()

        // Create and start WebSocket server
        val server = AudioWebSocketServer(
            port = port,
            path = path,
            streamManager = streamManager,
            memoryPool = memoryPool
        )

        // Handle graceful shutdown
        Runtime.getRuntime().addShutdownHook(Thread {
            Logger.info("Shutting down server...")
            server.stop()
        })

        Logger.info("Starting server... Press Ctrl+C to stop.")

        // Start server
        server.start()
    }
}
